Documentation/bootstrapping-guidelines.md

```diff
@@ -0,0 +1,96 @@
+
+To build .NET SDK from source requires previously source-built .NET SDK and toolset which is also referred as bootstrapping .NET SDK and toolset. Check quick [instructions](https://github.com/dotnet/installer/blob/main/README.md#build-net-from-source-source-build) to build SDK using source-build. This document provides guidance for getting or building  bootstrapping .NET SDK and toolset in different scenarios (covering step 2 and step 3 of source-build instructions).
+
+## Building New .NET Version
+Building New .NET Version is two stage process:  
+
+**Stage 1:**  
+Build bootstrapping .NET SDK and toolset for the targeted platform.
+- prep.sh (args???)  
+ This downloads platform-native version of .NET SDK and toolset. So if you are building on x64 machine, you get x64.
+- build.sh (args???)  
+Create source built .NET SDK
+
+**Stage 2:**  
+Use source built .NET SDK created in stage 1 to build .NET SDK from source. No need to run prep.sh in this stage.
+- Build.sh (Args - path to the source built SDK?????)
+
+## Building Subsequent .NET (Servicing)
+If you already have source-built .NET SDK and toolset in archive for the targeted platform and .NET version, then you do not need to recreate the bootstrapping .NET SDK and toolset. Subsequent versions will use the N-1 version toolset from the archive to build. Note, N-1 in this context means the previous version of .NET that was already built from source. As an example, to build SDK “N” (7.0.101) using “N-1” (7.0.100).
+
+To build .NET SDK from source -  
+ build.sh (args - path to the source built SDK???)
+
+## Building for New OS (Supported, RID Existing)
```

Comment:
What does "Supported, RID Existing" mean here?

Reply:
Added some details on this.